aws/resource_aws_batch_compute_environment_test.go

```diff
@@ -12,6 +15,93 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+func init() {
+	resource.AddTestSweepers("aws_batch_compute_environment", &resource.Sweeper{
+		Name: "aws_batch_compute_environment",
+		Dependencies: []string{
+			"aws_batch_job_queue",
+		},
+		F: testSweepBatchComputeEnvironments,
+	})
+}
+
+func testSweepBatchComputeEnvironments(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*AWSClient).batchconn
+
+	prefixes := []string{
+		"tf_acc",
+	}
+
+	out, err := conn.DescribeComputeEnvironments(&batch.DescribeComputeEnvironmentsInput{})
+	if err != nil {
+		return fmt.Errorf("Error retrieving Batch Compute Environments: %s", err)
+	}
+	for _, computeEnvironment := range out.ComputeEnvironments {
+		name := computeEnvironment.ComputeEnvironmentName
+		skip := true
+		for _, prefix := range prefixes {
+			if strings.HasPrefix(*name, prefix) {
+				skip = false
+				break
+			}
+		}
+		if skip {
+			log.Printf("[INFO] Skipping Batch Compute Environment: %s", *name)
+			continue
+		}
+
+		log.Printf("[INFO] Disabling Batch Compute Environment: %s", *name)
+
+		updateInput := &batch.UpdateComputeEnvironmentInput{
+			ComputeEnvironment: name,
+			State:              aws.String(batch.CEStateDisabled),
+		}
+		if _, err := conn.UpdateComputeEnvironment(updateInput); err != nil {
+			log.Printf("[ERROR] Failed to disable Batch Compute Environment %s: %s", *name, err)
+			continue
+		}
+
+		stateConfForDisable := &resource.StateChangeConf{
+			Pending:    []string{batch.CEStatusUpdating},
+			Target:     []string{batch.CEStatusValid},
+			Refresh:    resourceAwsBatchComputeEnvironmentStatusRefreshFunc(*name, conn),
+			Timeout:    20 * time.Minute,
+			MinTimeout: 5 * time.Second,
+		}
+		if _, err := stateConfForDisable.WaitForState(); err != nil {
+			log.Printf("[ERROR] Failed to wait for disable of Batch Compute Environment %s: %s", *name, err)
+			continue
+		}
+
+		log.Printf("[INFO] Deleting Batch Compute Environment: %s", *name)
+
+		deleteInput := &batch.DeleteComputeEnvironmentInput{
+			ComputeEnvironment: name,
+		}
+		if _, err := conn.DeleteComputeEnvironment(deleteInput); err != nil {
+			log.Printf("[ERROR] Failed to delete Batch Compute Environment %s: %s", *name, err)
+			continue
+		}
+
+		stateConfForDelete := &resource.StateChangeConf{
+			Pending:    []string{batch.CEStatusDeleting},
+			Target:     []string{batch.CEStatusDeleted},
+			Refresh:    resourceAwsBatchComputeEnvironmentDeleteRefreshFunc(*name, conn),
+			Timeout:    20 * time.Minute,
+			MinTimeout: 5 * time.Second,
+		}
+		if _, err := stateConfForDelete.WaitForState(); err != nil {
+			log.Printf("[ERROR] Failed to wait for deletion of Batch Compute Environment %s: %s", *name, err)
+		}
+	}
```

Comment:
I assume this is more or less copy of https://github.com/terraform-providers/terraform-provider-aws/blob/09d4f4ce96100b682a83cf3ae8bf589e6f36b65a/aws/resource_aws_batch_compute_environment.go#L307-L348 - which makes me wonder if we should move that logic into a function and reuse it here and in the Delete function?